modules/common-user-vpc/security_groups.tf

```diff
@@ -1,7 +1,7 @@
 resource "aws_security_group" "intra_vpc_ingress_and_egress" {
   description = "allow instances to talk to each other, and set up trusted ingress and egress"
   vpc_id      = var.vpc_id
-  name        = "${var.prefix}-intra-cluster-and-trusted-blocks"
+  name        = "${var.prefix}-${random_string.install_id.result}-intra-cluster-and-trusted-blocks"
```

Comment:
since there's a 255 character limit on this name, do we want to do something like this to truncate? or just let it fail on apply?


```hcl  
locals {
  sg_rendered_name = "${var.prefix}-${random_string.install_id.result}-intra-cluster-and-trusted-blocks"
  }
  
resource ... {
  name        = subtr(local.sg_rendered_name, 0, length(local.sg_rendered_name) < 255 ? length(local.sg_rendered_name) : 255
... }
```

Reply:
Hmmm, that's a really good thought! This is kind of a problem everywhere (though this has more characters in the base name), so I'm ok with addressing this separately at some point and letting the plan fail for now. 

Looking at a typical case here, the end user has about 200-220 characters to play with in the prefix, I feel like pushing back with that long of a prefix is ok and may help prevent less obvious collisions if someone went really long so everything had the same name from the truncation (though I'd love to see what that use case would look like 😂).

Still, something to be aware of while naming resources, glad you brought it up!